src/libraries/SparkLiquidityLayerHelpers.sol

```diff
@@ -0,0 +1,153 @@
+        uint256 depositMax,
+        uint256 depositSlope
+    ) internal {
+        IERC20 underlying = IERC20(IAToken(token).UNDERLYING_ASSET_ADDRESS());
+
+        RateLimitHelpers.setRateLimitData(
+            RateLimitHelpers.makeAssetKey(
+                LIMIT_AAVE_DEPOSIT,
+                token
+            ),
+            rateLimits,
+            RateLimitData({
+                maxAmount : depositMax,
+                slope     : depositSlope
+            }),
+            "atokenDepositLimit",
+            underlying.decimals()
+        );
+        RateLimitHelpers.setRateLimitData(
+            RateLimitHelpers.makeAssetKey(
+                LIMIT_AAVE_WITHDRAW,
+                token
+            ),
+            rateLimits,
+            RateLimitHelpers.unlimitedRateLimit(),
+            "atokenWithdrawLimit",
+            underlying.decimals()
+        );
+    }
+
+    /**
+     * @notice Onboard an ERC4626 vault
+     * @dev This will set the deposit to the given numbers with
+     *      the withdraw limit set to unlimited.
+     */
+    function onboardERC4626Vault(
+        address rateLimits,
+        address vault,
+        uint256 depositMax,
+        uint256 depositSlope
+    ) internal {
+        IERC20 asset = IERC20(IERC4626(vault).asset());
+
+        RateLimitHelpers.setRateLimitData(
+            RateLimitHelpers.makeAssetKey(
+                LIMIT_4626_DEPOSIT,
+                vault
+            ),
+            rateLimits,
+            RateLimitData({
+                maxAmount : depositMax,
+                slope     : depositSlope
+            }),
+            "vaultDepositLimit",
+            asset.decimals()
+        );
+        RateLimitHelpers.setRateLimitData(
+            RateLimitHelpers.makeAssetKey(
+                LIMIT_4626_WITHDRAW,
+                vault
+            ),
+            rateLimits,
+            RateLimitHelpers.unlimitedRateLimit(),
+            "vaultWithdrawLimit",
+            asset.decimals()
+        );
+    }
+
+    function morphoIdleMarket(
+        address asset
+    ) internal pure returns (MarketParams memory) {
+        return MarketParams({
+            loanToken:       asset,
+            collateralToken: address(0),
+            oracle:          address(0),
+            irm:             address(0),
+            lltv:            0
+        });
+    }
+
+    /**
+     * @notice Activate a Morpho Vault
+     * @dev This will do the following:
+     *      - Add the idle market for the underlying asset with unlimited size
+     *      - Add the relayer as an allocator
+     *      - Set the supply queue to the idle market
+     */
+    function activateMorphoVault(
+        address vault,
+        address relayer,
+        bool createIdleMarket
+    ) internal {
+        IERC20 asset = IERC20(IERC4626(vault).asset());
+        MarketParams memory idleMarket = morphoIdleMarket(address(asset));
+
+        if (createIdleMarket) {
+            IMorpho(MORPHO).createMarket(
```

Comment:
how does an idle market differ from markets we have onboarded in the past? why does the rule of 'not creating a market within a spell, instead create it beforehand  & seed it before referencing it in the spell' not apply for this case?

Reply:
Ah yes this is a fair point. This could be a race condition to make the spell fail. We should require the market be created in advance. I'm going to remove this conditional.

Reply:
Removed